File: src/p0048/golang/solution.go
func rotate(matrix [][]int) {
	horizontalFlip(matrix)
	diagonalFlip(matrix)
}

func horizontalFlip(matrix [][]int) {
	N := len(matrix)
	for i := 0; i < N/2; i++ {
		matrix[i], matrix[N-i-1] = matrix[N-i-1], matrix[i]
	}
}

func diagonalFlip(matrix [][]int) {
	N := len(matrix)
	for i := 0; i < N; i++ {
		for j := 0; j < i; j++ {
			matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]
		}
	}
}

File: src/p0048/cpp/solution.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
       auto n = matrix.size();
       int i = 0, j = n - 1;
       while (i < j) {
           swap(matrix[i++], matrix[j--]);
       }
       for (int i = 0; i < matrix.size(); i++) {
           for (int j = i; j < matrix[i].size(); j++) {
               swap(matrix[i][j], matrix[j][i]);
           }
       }
    }
};